### PR TITLE
List and Clear command feature

### LINKED ISSUE #52: List Appointment Feature

### LINKED ISSUE #53: Clear Appointment Feature

### PR DESCRIPTION
**Pull request by onnwards**
# Notes:
Fixes #52, Fixes #53 
Limited actual testing done for `list-appt` since we don't yet have a way to filter appointments

### Class Renaming: 
- `ListCommand` -> `ListPatientCommand`,
- `ClearCommand` -> `ClearPatientCommand`

### Command Words: 
- `list` -> `list-patient`
- `clear` -> `clear-patient`

### Add Typical Doctors, Timeslots, Appointments into `TypicalPersons.java`
- refactoring of `TypicalPersons.java` -> perhaps `TypicalAppObjects.java` will be done in another PR. 
- (too many files affected, will be very messy if done now)

### Add test method to display a specific appointment in `CommandTestUtil.java`, and predicate
- `showAppointmentAtIndex`
- method makes use of `AppointmentContainsKeywordsPredicate` which currently tests that an appointment's patient's name matches the keywords in the predicate. **Open to other ideas on how to better improve this implementation...**

## 3. Update files to use `getTypicalAppointmentSchedule()`
- was previously placeholder `new AppointmentScheduleBuilder().build()`
- future tests written should use `getTypicalAppointmentSchedule()` instead

## 4. Add Clear/List Appointment Command Feature
- `list-appt`, `clear-appt`
- Tests Written, `ListAppointmentCommandTest` and `ClearAppointmentCommandTest`
- Add `ListAppointmentCommand` and `ClearAppointmentCommand` Command Words into `UserInputParser`

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListAppointmentCommand.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_APPOINTMENTS;
+
+import seedu.address.model.Model;
+
+/**
+ * Lists all persons in the address book to the user.
```
Comment (Jacob-Pang) [inline src/main/java/seedu/address/logic/commands/ListAppointmentCommand.java]: Javadoc to update

File: src/main/java/seedu/address/logic/commands/ListPatientCommand.java [hunk 2]
```diff
@@ -8,11 +8,10 @@
 /**
  * Lists all persons in the address book to the user.
```
Comment (Jacob-Pang) [inline src/main/java/seedu/address/logic/commands/ListPatientCommand.java]: Javadoc to update

**Review by Jacob-Pang — APPROVED** [review]

File: src/test/java/seedu/address/logic/commands/ListPatientCommandTest.java [hunk 3]
```diff
@@ -4,39 +4,39 @@
...
     private Model model;
     private Model expectedModel;
 
     @BeforeEach
     public void setUp() {
-        model = new ModelManager(new AppointmentScheduleBuilder().build(), getTypicalAddressBook(),
+        model = new ModelManager(getTypicalAppointmentSchedule(), getTypicalAddressBook(),
                 new UserPrefs());
-        expectedModel = new ModelManager(new AppointmentScheduleBuilder().build(), model.getAddressBook(),
+        expectedModel = new ModelManager(getTypicalAppointmentSchedule(), model.getAddressBook(),
```
Comment (Jacob-Pang) [inline src/test/java/seedu/address/logic/commands/ListPatientCommandTest.java:28]: This one should be getAppointmentSchedule instead

Comment (onnwards) [reply src/test/java/seedu/address/logic/commands/ListPatientCommandTest.java:28]: don't think so, logic is we don't want any patient commands to affect appointment schedule, so we make a new typicalAppointmentSchedule here

**Pull request merged**